src/flint/fmpz_mod_poly.jl

```diff
@@ -777,24 +777,27 @@ function fit!(x::T, n::Int) where {T <: Zmodn_fmpz_poly}
   return nothing
 end
 
-function setcoeff!(x::T, n::Int, y::UInt) where {T <: Zmodn_fmpz_poly}
-  @ccall libflint.fmpz_mod_poly_set_coeff_ui(x::Ref{T}, n::Int, y::UInt, x.parent.base_ring.ninv::Ref{fmpz_mod_ctx_struct})::Nothing
-  return x
+#
+
+function setcoeff!(z::T, i::Int, x::ZZRingElemOrPtr, ctx::fmpz_mod_ctx_struct=base_ring(z).ninv) where T <: Zmodn_fmpz_poly
```

Comment:
Should `ctx` be another (fourth) argument or a kwarg? 

The advantage of a kwarg is that it helps understanding what's going on, esp. in the cases were `ctx` is another integer. If I see `set!(z, 3, c, R)` it is somewhat more opaque than `set!(z, 3, c; ctx=R)`.

Alas, in the end, all of these are low-level unsafe helpers. I can happily live with either variant.